packages/frontend/src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -1,32 +1,125 @@
 import { selectActionsPending, selectStatusSlice } from '../../../redux/slices/status';
 import { WalletError } from '../../../utils/walletError';
+import AlertBanner from '../../common/AlertBanner';
 import FormButton from '../../common/FormButton';
 import Modal from '../../common/modal/Modal';
 import ModalTheme from '../ledger/ModalTheme';
 import TwoFactorVerifyInput from './TwoFactorVerifyInput';
 
+const {
+    format: {
+        formatNearAmount
+    },
+    key_pair: {
+        PublicKey,
+        KeyType
+    },
+} = utils;
+
 const Form = styled.form`
     display: flex;
     flex-direction: column;
     align-items: center;
     width: 100%;
 `;
 
+const formatNear = (amount) => {
+    try {
+        return formatNearAmount(amount, 4);
+    } catch {
+        // Near amount may be stored as hex rather than decimal
+        return formatNearAmount(new BN(amount, 16).toString(), 4);
+    }
+};
+
+const getTranslationsFromMultisigRequest = ({ actions, receiverId, accountId }) => {
+    const fullAccessKeyAction = actions.find(({ enum: type, permission }) => type === 'addKey' && !permission);
+    if (fullAccessKeyAction) {
+        const publicKey = new PublicKey({
+            keyType: KeyType.ED25519,
+            data: fullAccessKeyAction.addKey.publicKey.data
+        });
+        return [
+            {
+                id: 'twoFactor.action.addKey.full',
+                data: {
+                    accountId,
+                    publicKey: publicKey.toString(),
+                }
+            }
+        ];
+    }
+
+    return actions
+        .map(({ enum: actionType, [actionType]: action }) => {
+            switch (actionType) {
+                case 'addKey':
+                    return {
+                        id: 'twoFactor.action.addKey.limited',
+                        data: {
+                            receiverId,
+                            methodNames: action.permission.functionCall.methodNames.join(', '),
+                            allowance: formatNearAmount(action.permission.functionCall.allowance, 4),
+                            publicKey: new PublicKey({ keyType: KeyType.ED25519, data: action.publicKey.data }).toString(),
+                        }
+                    };
+                case 'deleteKey':
+                    return {
+                        id: 'twoFactor.action.deleteKey',
+                        data: {
+                            publicKey: new PublicKey({ keyType: KeyType.ED25519, data: action.publicKey.data }).toString(),
+                        }
+                    };
+                case 'functionCall':
+                    return {
+                        id: 'twoFactor.action.functionCall',
+                        data: {
+                            receiverId,
+                            methodName: action.methodName,
+                            deposit: formatNear(action.deposit, 4),
```

Comment:
Think we can cut the `4` here since the second param isn't used in the `formatNear` function

Reply:
Woops, left behind from when this was `formatNearAmount` 😅